breadbox/breadbox/schemas/compute.py

```diff
@@ -1,20 +1,41 @@
 from pydantic import BaseModel, Field
-from typing import Optional, Any, List
+from typing import Optional, Annotated, Any, Literal
 
 
 class ComputeParams(BaseModel):
-    analysisType: str
-    datasetId: str
-    # Either queryId (node id) or queryFeatureId and queryDatasetId are required
-    # queryId is a deprecated query type which is only used by Elara
-    queryId: Optional[str] = None
-    queryFeatureId: Optional[str] = None
-    queryDatasetId: Optional[str] = None
-    vectorVariableType: Optional[str] = None
-    queryCellLines: Optional[List[str]] = None
-    queryValues: Optional[List[Any]] = (
-        None  # The query may be specified with either the QueryId or QueryValues
-    )
+    analysisType: Annotated[
+        Literal["pearson", "association", "two_class"],
+        Field(description="The type of analysis to run")
+    ]
+    datasetId: Annotated[
+        str, Field(description="The given ID or UUID of the dataset to search for association")
```

Comment:
I love the addition of these annotations! I feel like this will clear up a lot of confusion. I used to get so confused about which id is the dataset being searched, and which is the data slice you're querying for...